packages/relay-compiler/language/RelayLanguagePluginInterface.js

```diff
@@ -0,0 +1,246 @@
+   */
+  typeText: string,
+
+  /**
+   * The name of the relay-runtime module being used.
+   */
+  relayRuntimeModule: string,
+
+  /**
+   * A hash of the concrete node including the query text.
+   *
+   * @todo Document how this is different from `sourceHash`.
+   */
+  hash: ?string,
+
+  /**
+   * A hash of the document, which is used by relay-compiler to know if it needs
+   * to write a new version of the artifact.
+   *
+   * @todo Is this correct? And document how this is different from `hash`.
+   */
+  sourceHash: string,
+
+  /**
+   * @todo Document this.
+   */
+  devOnlyAssignments: ?string,
+|}) => string;
+
+/**
+ * The options that will be passed to the `generate` function of your plugin’s
+ * type generator.
+ */
+export type TypeGeneratorOptions = {|
+  /**
+   * A map of custom scalars to scalars that the plugin knows about and emits
+   * tpye information for.
+   *
+   * @example
+   *
+   *  // The URL custom scalar is essentially a string and should be treated as
+   *  // such by the language’s type system.
+   *  { URL: 'String' }
+   */
+  +customScalars: {[type: string]: string},
+
+  /**
+   * Lists all other fragments relay-compiler knows about. Use this to know when
+   * to import/reference other artifacts.
+   */
+  +existingFragmentNames: Set<string>,
+
+  /**
+   * The name of the relay-runtime module.
+   *
+   * This defaults to `relay-runtime`.
+   */
+  +relayRuntimeModule: string,
+
+  /**
+   * Whether or not relay-compiler will store artifacts next to the module that
+   * they originate from or all together in a single directory.
+   *
+   * Storing all artifacts in a single directory makes it easy to import and
+   * reference fragments defined in other artifacts without needing to use the
+   * Haste module system.
+   *
+   * This defaults to `false`.
+   */
+  +useSingleArtifactDirectory: boolean,
+
+  /**
+   * @todo Document this.
+   */
+  +inputFieldWhiteList: $ReadOnlyArray<string>,
+
+  /**
+   * Whether or not the Haste module system is being used. This will currently
+   * always be `false` for OSS users.
+   */
+  +useHaste: boolean,
+
+  /**
+   * @todo Document this.
+   */
+  +enumsHasteModule: ?string,
+|};
+
+/**
+ * This object should hold the implementation required to generate types for the
+ * GraphQL selections made.
+ *
+ * @see {@link javascript/RelayFlowGenerator.js}
+ */
+export type TypeGenerator = {
+  /**
+   * Transforms that should be applied to the intermediate representation of the
+   * GraphQL document before passing to the `generate` function.
+   */
+  transforms: Array<IRTransform>,
+
+  /**
+   * Given GraphQL document IR, this function should generate type information
+   * for e.g. the selections made. It can, however, also generate any other
+   * content such as importing other files, including other artifacts.
+   */
+  generate: (node: Root | Fragment, options: TypeGeneratorOptions) => string,
+};
```

Comment:
all of these types just got moved here right? did they also have all these comments? if the didn't, thanks for adding them!

Reply:
These all moved here indeed, but they did not have these comments 😄